tests/api/test_api.py

```diff
@@ -88,16 +90,32 @@ def get_api_test_info(path):
 
         output_modifier = globals()[modifier_name]
 
-    return import_module_name, arguments, data['output'], output_modifier
+    return import_module_name, arguments, data['output'], is_error, output_modifier
 
 def _get_api_test_method(path):
-    import_module_name, arguments, expected, output_modifier = get_api_test_info(path)
+    import_module_name, arguments, expected, is_error, output_modifier = get_api_test_info(path)
 
     def __method(self):
         api_module = importlib.import_module(import_module_name)
 
         self._next_response_queue.put(expected)
-        actual = api_module.send(arguments)
+
+        try:
+            actual = api_module.send(arguments)
+        except Exception as ex:
+            if (not is_error):
+                raise ex
+
+            python_message = expected.get('python-message', "")
+            self.assertEqual(python_message, str(ex))
+
+            code = expected.get('code', 0)
+            self.assertEqual(code, ex.code)
+            return
```

Comment:
I would put a newline before this return.
This gives better symmetry with the above vertical group 
(both fetch something and then assert).

Note that this is a tiny nit and subjective
(so small that I would not include it if there were not other things in the PR).